src/main/java/org/tikv/common/log/SlowLogImpl.java

```diff
@@ -68,45 +62,53 @@ public synchronized SlowLogSpan start(String name) {
     return slowLogSpan;
   }
 
+  @Override
+  public long getTraceId() {
+    return traceId;
+  }
+
+  @Override
+  public long getThresholdMS() {
+    return slowThresholdMS;
+  }
+
   @Override
   public void setError(Throwable err) {
     this.error = err;
   }
 
   @Override
   public void log() {
+    if (error != null || timeExceeded()) {
+      logger.warn("SlowLog:" + getSlowLogJson());
+    }
+  }
+
+  boolean timeExceeded() {
     long currentNS = System.nanoTime();
-    long currentMS = startMS + (currentNS - startNS) / 1_000_000;
-    if (error != null || (slowThresholdMS >= 0 && currentMS - startMS > slowThresholdMS)) {
-      logger.warn("SlowLog:" + getSlowLogString(currentMS));
+    long durationMS = (currentNS - startNS) / 1_000_000;
+    if (slowThresholdMS >= 0 && durationMS > slowThresholdMS) {
+      logger.warn("A slow request spent " + durationMS + "ms");
```

Comment:
it is duplicated with the log printed by the callers, is it necessary to print this log?

Reply:
This is aimed to print a human-friendly total time in ms, while the JSON is considered to be machine-friendly.

Reply:
we should limit the log amount as less as possible. please only print one slowlog for one request.